js/views/search/Search.js

```diff
@@ -417,177 +412,172 @@ export default class extends baseVw {
-        isDefaultProvider,
-        emptyData,
+        providerLocked: this.providerIsADefault(this._search.provider.id),
+        isExistingProvider: this.isExistingProvider(this._search.provider),
+        showMakeDefault: this._search.provider !== this.currentDefaultProvider,
+        showDataError: $.isEmptyObject(data) && !state.showHome,
+        hasFilters,
         ...state,
-        ...this.sProvider,
         ...data,
       }));
     });
-    this.$sortBy = this.$('#sortBy');
-    this.$sortBy.select2({
-      // disables the search box
-      minimumResultsForSearch: Infinity,
-      templateResult: selectEmojis,
-      templateSelection: selectEmojis,
-    });
-    const filterWrapper = this.$('.js-filterWrapper');
-    filterWrapper.find('select').select2({
-      minimumResultsForSearch: 10,
-      templateResult: selectEmojis,
-      templateSelection: selectEmojis,
-    });
-    this.$filters = filterWrapper.find('select, input');
-    this.$resultsWrapper = this.$('.js-resultsWrapper');
-    this.$searchInput = this.$('.js-searchInput');
-    this.$searchLogo = this.$('.js-searchLogo');
 
-    this.$searchLogo.find('img').on('error', () => {
-      this.$searchLogo.addClass('loadError');
+    const $filterWrapper = this.$('.js-filterWrapper');
+    const $searchLogo = this.$('.js-searchLogo');
+
+    $searchLogo.find('img').on('error', () => {
+      $searchLogo.addClass('loadError');
     });
 
     if (this.searchProviders) this.searchProviders.remove();
     this.searchProviders = this.createChild(Providers, {
-      urlType: this.urlType,
-      currentID: this.getCurrentProviderID(),
-      selecting: this.mustSelectDefault,
+      searchType: this._search.searchType,
+      currentID: this._search.provider.id,
+      showSelectDefault: !this.currentDefaultProvider,
     });
     this.listenTo(this.searchProviders, 'activateProvider', pOpts => this.activateProvider(pOpts));
     this.$('.js-searchProviders').append(this.searchProviders.render().el);
 
     if (this.suggestions) this.suggestions.remove();
-    this.suggestions = this.createChild(Suggestions, {
-      initialState: {
-        suggestions: Array.isArray(data.suggestions) ? data.suggestions : this.defaultSuggestions,
-      },
-    });
+    this.suggestions = this.createChild(Suggestions);
     this.listenTo(this.suggestions, 'clickSuggestion', opts => this.onClickSuggestion(opts));
     this.$('.js-suggestions').append(this.suggestions.render().el);
 
-    // use the initial set of results data to create the results view
-    if (data) this.createResults(data, state.searchUrl);
+    this.categoryViews.forEach(cat => cat.remove());
+    if (this.filters) this.filters.remove();
+    if (this.sortBy) this.sortBy.remove();
+
+    if (state.showHome) {
+      // Create a disposable copy to be shifted by the addNextCategory function.
+      this._categorySearchesToAdd = [...this._categorySearches];
+      this.addNextCategory();
```

Comment:
Not suggesting some massive refactor at this point... but (I've banged this drums a few times over the years... lol), it's pretty much an anti-pattern to kick off side effects in render. You have to be very careful. For example, with this structured as it is, if you do:

setState({ showHome: true, somethingLabel: 'Hi There' }) // triggers a category fetch
// while the category is fetching -or- some subsequent one that addNextCategory kicked off, you need to update some state, eg.
setState({ showHome: true, somethingLabel: 'Bye Bye!' }) // triggers a another add category fetch and now you have to addCategoryNext cycles going

You've basically made it that any state change or render will result in a fetch regardless of if that state change or render has anything to do with the fetch.